src/main/kotlin/com/openlattice/chronicle/services/jobs/JobService.kt

```diff
@@ -52,33 +61,67 @@ class JobService (
         private val GET_JOBS_SQL = """
             SELECT * FROM ${JOBS.name} WHERE ${JOB_ID.name} = ANY(?)
         """.trimIndent()
+
+        private val GET_NEXT_JOB_SQL = """
+            DELETE FROM ${JOBS.name}
+            WHERE ${JOB_ID.name} = (
+                SELECT ${JOB_ID.name}
+                FROM ${JOBS.name}
+                ORDER BY ${PostgresColumns.CREATED_AT.name} ASC
+                FOR UPDATE SKIP LOCKED
+                LIMIT 1
+            )
+            RETURNING *
+        """.trimIndent()
```

Comment:
Won't this delete the job once it's done even after it's been marked with `status = 'FINISHED'`.

Reply:
Good point!

Reply:
fixed and added an hourly query to clean out jobs older than 1 week.